File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        prefix_sum = 0
        mp = {0: 1}

        for num in nums:
            prefix_sum += num
            diff = prefix_sum - k
            count += mp.get(diff, 0)

            mp[prefix_sum] = mp.get(prefix_sum, 0) + 1

        return count

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int, int> mp;
        mp[0] = 1;

        int prefixSum = 0;
        int count = 0;

        for (int num : nums) {
            prefixSum += num;

            count += mp[prefixSum - k];

            mp[prefixSum]++;
        }

        return count;
    }
};